examples/05_deduplication.py

```diff
@@ -1,66 +1,67 @@
 
-Deduplication relies on *unsupervised learning*, to find structure in
-data without providing explicit labels/categories of the data a-priori.
-Specifically clustering of the distance between strings can be used to
+Merging multiple variants of the same category is known as
+*deduplication*. It is implemented in skrub with the |deduplicate| function.
+
+Deduplication relies on *unsupervised learning*, it finds structures in
+the data without providing a-priori known and explicit labels/categories.
+Specifically, measuring the distance between strings can be used to
 find clusters of strings that are similar to each other (e.g. differ only
-by a misspelling) and hence gives us an easy tool to flag potentially
+by a misspelling) and hence, flag and regroup potentially
 misspelled category names in an unsupervised manner.
 
 
-.. |dd| replace:: :func:`~skrub.deduplicate`
+.. |deduplicate| replace::
+    :func:`~skrub.deduplicate`
+
+ .. |Gap| replace::
+     :class:`~skrub.GapEncoder`
+
+ .. |MinHash| replace::
+     :class:`~skrub.MinHashEncoder`
 """
 
 ###############################################################################
 # An example dataset
-# ------------------
+# -------------------
 #
-# Imagine the following example:
-# As a data scientist, our job is to analyze the data from a hospital ward.
-# We notice that most of the cases involve the prescription of one of three different
-# medications: "Contrivan", "Genericon", or "Zipholan".
-# However, data entry is manual and - either because the prescribing doctor's
-# handwriting was hard to decipher, or due to mistakes during data input - there are
-# multiple spelling mistakes for these three medications.
+# Let's take an example:
+# as a data scientist, your job is to analyze the data from a hospital ward.
+# In the data, we notice that in most of the cases the doctor prescribes
+# one of three following medications:
+# "Contrivan", "Genericon", or "Zipholan".
 #
-# Let's generate some example data that demonstrate this.
-
+# However, data entry is manual and - either because the doctor's
+# handwriting was hard to decipher, or due to mistakes during input -
+# there are multiple spelling mistakes in the dataset.
+#
+# Let's generate this example dataset:
 
+import pandas as pd
 import numpy as np
-
 from skrub.datasets import make_deduplication_data
 
-# our three medication names
-medications = ["Contrivan", "Genericon", "Zipholan"]
-entries_per_medications = [500, 100, 1500]
-
-# 5% probability of a typo per letter
-prob_mistake_per_letter = 0.05
-
 duplicated_names = make_deduplication_data(
-    medications,
-    entries_per_medications,
-    prob_mistake_per_letter,
+    examples=["Contrivan", "Genericon", "Zipholan"],  # our three medication names
+    entries_per_example=[500, 100, 1500],  # their respective number of occurrences
+    prob_mistake_per_letter=0.05,  # 5% probability of typo per letter
     random_state=42,  # set seed for reproducibility
 )
```

Comment:
Can you display the generated data (the "duplicated_names") and add a cell break here, please